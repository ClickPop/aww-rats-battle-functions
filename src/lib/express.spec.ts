import request from 'supertest';
import { app } from '..';

const agent = request.agent(app);

describe('/', () => {
  test('should return available routes with correct api key', async () => {
    const res = await agent.get('/').set('authorization', 'test');
    expect(res.statusCode).toEqual(200);
    expect(res.body).toMatchSnapshot();
  });

  test('should error if given incorrect api key', async () => {
    const res = await agent.get('/').set('authorization', 'wrong');
    expect(res.statusCode).toEqual(401);
    expect(res.body).toMatchSnapshot();
  });

  test('should error if no api key is supplied', async () => {
    const res = await agent.get('/');
    expect(res.statusCode).toEqual(401);
    expect(res.body).toMatchSnapshot();
  });
});
