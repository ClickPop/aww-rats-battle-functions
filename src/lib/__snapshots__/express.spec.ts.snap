// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ should error if given incorrect api key 1`] = `
Object {
  "error": "invalid api key on request",
}
`;

exports[`/ should error if no api key is supplied 1`] = `
Object {
  "error": "invalid api key on request",
}
`;

exports[`/ should return available routes with correct api key 1`] = `
Array [
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/auth/login",
  },
  Object {
    "methods": Array [
      "get",
    ],
    "path": "/auth",
  },
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/auth-check",
  },
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/caching/clear",
  },
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/solo-encounter-attempt",
  },
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/add-raid-contribution",
  },
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/cron/reset-energy",
  },
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/cron/handle-raid-trigger",
  },
  Object {
    "methods": Array [
      "post",
    ],
    "path": "/cron/handle-raid-end",
  },
]
`;
